pkg/apis/core/validation/validation.go

```diff
@@ -1564,36 +1564,44 @@ func validateCSIPersistentVolumeSource(csi *core.CSIPersistentVolumeSource, fldP
 		if len(csi.ControllerPublishSecretRef.Name) == 0 {
 			allErrs = append(allErrs, field.Required(fldPath.Child("controllerPublishSecretRef", "name"), ""))
 		} else {
-			allErrs = append(allErrs, ValidateDNS1123Label(csi.ControllerPublishSecretRef.Name, fldPath.Child("name"))...)
+			for _, msg := range ValidateSecretName(csi.ControllerPublishSecretRef.Name, false) {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("controllerPublishSecretRef"), csi.ControllerPublishSecretRef.Name, msg))
```

Comment:
The invalid field is `ControllerPublishSecretRef.Name`, therefore `fldPath.Child("controllerPublishSecretRef", "name")`

(this applies to all similar cases below)

Reply:
ok.. sure.. thanks.